src/dfa.js

```diff
@@ -410,35 +410,24 @@ let generateDFA = (roles, i, n, k, sets, implied) => {
 exports.generateDFA = generateDFA;
 
 /**
- * Builds a pair {resolver, sets} where sets is a list of lists of roles,
- * and given a scope `resolver(scope)`` returns an index from sets.
- *
- * That definition sounds slightly complicated, it's actually very simple,
- * sets is on the form:
- * ```js
- * sets = [
- *   [{role}, ...],
- *   [{role}, ...],
- *   [{role}, ...]
- * ]
- * ```
- *
- * For efficiency we allow sets[i] = [{role}, ..., j] where j < i to be
- * interpreted as sets[i].concat(sets[j]). By not duplicating we don't have to
- * resolve `sets[j]` multiple times, even though the set appears in multiple
- * other sets.
- *
- * The `resolver` function returns an index in the sets array, as this allows
- * us to later create a new sets variable where roles have been expanded to
- * the scopes they imply, and just like that we can use the `resolver` to go
- * from scope to expanded scopes.
+ * Finds the set identifier given a scope and initial state.
  */
-let buildResolver = (roles) => {
-  // Generate DFA
-  roles = sortRolesForDFAGeneration(roles);
-  let sets = [[]];
-  let dfa = generateDFA(roles, 0, roles.length, 0, sets, 0);
+let executeDFA = (state, scope, depth = 0) => {
+  let n = scope.length;
+  if (typeof(state.end) === 'number' && n == depth) {
```

Comment:
Maybe add a comment here about why we have to check if it's a number. It's because of the case where the initial character isn't a match at all, right? It is kinda non-obvious.